pageserver/src/tenant.rs

```diff
@@ -1138,7 +1146,8 @@ impl Tenant {
                                 )
                                 .context("create_timeline_struct")?;
 
-                            let guard = Arc::clone(&timeline.delete_lock).lock_owned().await;
+                            let guard =
+                                DeletionGuard(Arc::clone(&timeline.delete_lock).lock_owned().await);
```

Comment:
Not needed in this PR, but we should use `try_lock_owned().expect()` We should use it here, because, we just created the timeline struct, it's owned by us still.

Reply:
You mean try_lock_owned().expect("shouldnt happen because we're a single owner")?

Reply:
Pushed https://github.com/neondatabase/neon/pull/4495/commits/2541f432595de24926b503435c0500329d9dd9b6